packages/led_driver/include/rgb_led/virtual_rgb_led.py

```diff
@@ -0,0 +1,141 @@
+
+    Low level class that creates the PWM messages that are sent to the
+    microcontroller. It contains offset addresses relatives to the
+    address of the various LEDs.
+
+    Each LED on a Duckiebot or a watchtower is indexed by a number:
+
+    +------------------+------------------------------------------+
+    | Index            | Position (rel. to direction of movement) |
+    +==================+==========================================+
+    | 0                | Front left                               |
+    +------------------+------------------------------------------+
+    | 1                | Rear left                                |
+    +------------------+------------------------------------------+
+    | 2                | Top / Front middle                       |
+    +------------------+------------------------------------------+
+    | 3                | Rear right                               |
+    +------------------+------------------------------------------+
+    | 4                | Front right                              |
+    +------------------+------------------------------------------+
+
+    Setting the color of a single LED is done by setting the brightness of the
+    red, green, and blue channels to a value between 0 and 255. The communication
+    with the hardware controller is abstracted through the :obj:`setRGB` method. By
+    using it, you can set directly set the desired color to any LED.
+
+    """
+
+    def __init__(self, debug=False):
+        
+        self._matrix: Optional[Matrix] = None
+        self._device: Optional[Lights] = None
+        # register connection setup function
+        self._connection_request: Optional[DuckiematrixLinkDescription] = None
+        on_duckiematrix_connection_request(self.on_connection_request)
+    
+    def on_connection_request(self, link: DuckiematrixLinkDescription):
+        class_name = self.__class__.__name__
+        # store new connection request
+        self._connection_request = link
+        print(f"[{class_name}]: executing connection request")
+        # Setup the connection to the LEDs
+        self.release()
+        print(f"[{class_name}]: Pre-setup")
+        self.setup()
+        print(f"[{class_name}]: Post-setup")
+
+
+    def setup(self):
+        if self._connection_request is None:
+            print("Not connecting")
+            return
+        # ---
+        class_name = self.__class__.__name__
+        print(f"[{class_name}] Executing virtual led setup: {self._connection_request}")
+        link = self._connection_request
+        configuration = get_robot_configuration()
+        # create connection to the matrix engine
+        self._matrix: Matrix = Matrix(link.uri,auto_commit=True)
+        print(f"[{class_name}] connected to Matrix at {link.uri}")
+        # create connection to the vehicle
+        robot: DifferentialDriveRobot = self._matrix.robots.create(configuration.name, link.entity)
```

Comment:
do not use `matrix.robots.create(...)` here, use `matrix.LightsEnabledRobot(...)` instead.